PurchasesTests/Caching/DeviceCacheTests.swift

```diff
@@ -198,31 +198,46 @@ class DeviceCacheTests: XCTestCase {
         ])
         guard let offering = optionalOffering else { fatalError("couldn't create offering for tests") }
         let expectedOfferings = Offerings(offerings: ["offering1": offering], currentOfferingID: "base")
-        self.deviceCache.cacheOfferings(expectedOfferings)
+        self.deviceCache.cache(offerings: expectedOfferings)
 
         expect(self.deviceCache.cachedOfferings).to(beIdenticalTo(expectedOfferings))
     }
 
-    func testCrashesWhenAppUserIDIsDeleted() {
+
+    func testAssertionHappensWhenAppUserIDIsDeleted() {
         let mockNotificationCenter = MockNotificationCenter()
         mockUserDefaults.mockValues["com.revenuecat.userdefaults.appUserID.new"] = "Rage Against the Machine"
 
-        self.deviceCache = RCDeviceCache(mockUserDefaults,
-                                         offeringsCachedObject: nil,
-                                         notificationCenter: mockNotificationCenter)
+        #if arch(arm64)
+        let assertionHappened = expectation(description: "Assertion happened")
+        self.deviceCache = DeviceCache(userDefaults: mockUserDefaults,
+                                       offeringsCachedObject: nil,
+                                       notificationCenter: mockNotificationCenter,
+                                       assertionFunction: { _ in assertionHappened.fulfill() })
+        #else
+        self.deviceCache = DeviceCache(userDefaults: mockUserDefaults,
+                                       offeringsCachedObject: nil,
+                                       notificationCenter: mockNotificationCenter)
```

Comment:
why is this needed? does `toEventually` not do the trick here?

Reply:
negative, can't test assertions in in `arm64` that way 😢